pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -491,13 +493,22 @@ impl GcCompactionQueue {
                         let mut guard = self.inner.lock().unwrap();
                         guard.guards.entry(id).or_default().gc_guard = Some(gc_guard);
                     }
-                    let _ = timeline.compact_with_options(cancel, options, ctx).await?;
+                    let compaction_result =
+                        timeline.compact_with_options(cancel, options, ctx).await?;
                     self.notify_and_unblock(id);
+                    if compaction_result == CompactionOutcome::YieldForL0 {
+                        yield_for_l0 = true;
```

Comment:
nit: why not just return early here, instead of using `yield_for_l0`?

Reply:
We have some cleanup work in the end before returning from this function, will add some comments

Reply:
Ah, I missed the `guard.running = None;`. We could use a `defer!` for that, but 🤷.